openapi-spec/swagger.yaml

```diff
@@ -1287,6 +1287,296 @@ paths:
+  '/v1beta/{tenantId}/file/shares/{fileshareId}':
+    parameters:
+      - $ref: '#/parameters/tenantId'
+      - $ref: '#/parameters/fileshareId'
+    get:
+      tags:
+        - File Shares
+      description: Gets fileshare detail by fileshare id.
+      responses:
+        '200':
+          description: OK
+          schema:
+            $ref: '#/definitions/FileShareSpec'
+        '401':
+          $ref: '#/responses/HTTPStatus401'
+        '403':
+          $ref: '#/responses/HTTPStatus403'
+        '404':
+          $ref: '#/responses/HTTPStatus404'
+        '500':
+          $ref: '#/responses/HTTPStatus500'
+    put:
+      tags:
+        - File Shares
+      description: Updates a file share.
+      parameters:
+        - name: fileshare
+          in: body
+          schema:
+            type: object
+            required:
+              - name
+            properties:
+              name:
+                type: string
+              description:
+                type: string
+      responses:
+        '200':
+          description: OK
+          schema:
+            $ref: '#/definitions/FileShareSpec'
+        '401':
+          $ref: '#/responses/HTTPStatus401'
+        '403':
+          $ref: '#/responses/HTTPStatus404'
+        '500':
+          $ref: '#/responses/HTTPStatus500'
+    delete:
+      tags:
+        - File Shares
+      description: Deletes a fileshare.
+      responses:
+        '202':
+          description: Accepted
+        '401':
+          $ref: '#/responses/HTTPStatus401'
+        '403':
+          $ref: '#/responses/HTTPStatus403'
+        '404':
+          $ref: '#/responses/HTTPStatus404'
+        '500':
+          $ref: '#/responses/HTTPStatus500'
+  '/v1beta/{tenantId}/file/snapshots':
+    parameters:
+      - $ref: '#/parameters/tenantId'
+    get:
+      tags:
+        - File Share Snapshots
+      description: Lists information for all file share spanshots.
+      responses:
+        '200':
+          description: OK
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/FileShareSnapshotSpec'
+        '401':
+          $ref: '#/responses/HTTPStatus401'
+        '403':
+          $ref: '#/responses/HTTPStatus403'
+        '500':
+          $ref: '#/responses/HTTPStatus500'
+    post:
+      tags:
+        - File Share Snapshots
+      description: Creates a file share snapshot.
+      parameters:
+        - name: body
+          in: body
+          schema:
+            $ref: '#/definitions/FileShareSnapshotSpec'
+      responses:
+        '202':
+          description: Accepted
+          schema:
+            $ref: '#/definitions/FileShareSnapshotSpec'
+        '400':
+          $ref: '#/responses/HTTPStatus400'
+        '401':
+          $ref: '#/responses/HTTPStatus401'
+        '403':
+          $ref: '#/responses/HTTPStatus403'
+        '500':
+          $ref: '#/responses/HTTPStatus500'
+  '/v1beta/{tenantId}/file/snapshots/{fileshareId}':
```

Comment:
File share could have multiple snapshot at the same time, so please use `shareSnapshotId` here.

Reply:
Yes file share supports multiple snapshots, and they can create using existing fileshareId so source resource is fileshareId, how it will be suitable if we replace fileshareId with shareSnapshotId.

Reply:
You mean should i have to add one more api for creating snapshotId for existing snapshot?

Reply:
No, just replace `shareSnapshotId` with `fileshareId`

Reply:
ok i got it. to update, get, delete i should add sharesnapshotId. Done.